Mobile/ShopLens/Droid/Activities/ShoppingListActivity.cs

```diff
@@ -68,39 +95,57 @@ private bool IsTalkBackEnabled()
 
         private void InitiateNoTalkBackMode()
         {
-            tts = new TextToSpeech(this, this);
+            needUserAnswerId = ConfigurationManager.AppSettings["AnswerUtteranceId"];
+            askUserToRepeat = ConfigurationManager.AppSettings["AskToRepeat"];
+            afterActionAsk = ConfigurationManager.AppSettings["AfterActionMsg"];
+            cmdOpenMain = ConfigurationManager.AppSettings["CmdOpenMain"];
+            cmdOpenCart = ConfigurationManager.AppSettings["CmdOpenCart"];
+            cmdVoiceList = ConfigurationManager.AppSettings["CmdVoiceCartList"];
+            cmdHelp = ConfigurationManager.AppSettings["CmdHelp"];
+            cmdRemind = ConfigurationManager.AppSettings["CmdRemind"];
+
+            shopLensTTS = new ShopLensTextToSpeech(this, TTSSpeakAfterInit, TTStoppedSpeaking);
+            voiceRecognizer = new ShopLensSpeechRecognizer(OnVoiceRecognitionResults);
         }
 
-        public void OnInit([GeneratedEnum] OperationResult status)
+        private void TTSSpeakAfterInit(object sender, EventArgs e)
         {
-            if (status == OperationResult.Error)
-            {
-                tts.SetLanguage(Locale.Default);
-            }
-            else if (status == OperationResult.Success)
+            var message = ConfigurationManager.AppSettings["ListOnRestartMsg"];
+            shopLensTTS.Speak(message, needUserAnswerId);
+        }
+
+        private void TTStoppedSpeaking(object sender, UtteranceIdArgs e)
```

Comment:
```suggestion
        private void TTSStoppedSpeaking(object sender, UtteranceIdArgs e)
```

Or, another recommended way of writing acronyms in camel case, `TtsStoppedSpeaking`